test/test_unidler.py

```diff
@@ -0,0 +1,253 @@
+from io import BytesIO
+from unittest.mock import MagicMock, call, patch
+
+import pytest
+
+import unidler
+from unidler import (
+    IDLED,
+    IDLED_AT,
+    INGRESS_CLASS,
+    RequestHandler,
+    UNIDLER,
+)
+
+
+HOSTNAME = 'test.host.name'
+
+
+@pytest.yield_fixture
+def client():
+    client = MagicMock()
+    with patch('unidler.client', client):
+        yield client
+
+
+@pytest.fixture
+def deployment():
+    deployment = MagicMock()
+    deployment.spec.replicas = 0
+    deployment.metadata.labels = {
+        IDLED: 'true',
+    }
+    deployment.metadata.annotations = {
+        IDLED_AT: 'YYYY-mm-ddTHH:MM:SS+0000,2',
+    }
+    deployment.metadata.name = 'test-app'
+    deployment.metadata.namespace = 'test-namespace'
+    return deployment
+
+
+@pytest.fixture
+def ingress():
+    host_rule = MagicMock()
+    host_rule.host = HOSTNAME
+    ingress = MagicMock()
+    ingress.spec.rules = [
+        host_rule,
+    ]
+    ingress.metadata.annotations = {
+        INGRESS_CLASS: 'disabled',
+    }
+    ingress.metadata.name = 'test-app'
+    ingress.metadata.namespace = 'test-namespace'
+    return ingress
+
+
+@pytest.fixture
+def unidler_ingress():
+    host_rule = MagicMock()
+    host_rule.host = HOSTNAME
+    ingress = MagicMock()
+    ingress.spec.rules = [
+        host_rule,
+    ]
+    ingress.metadata.name = UNIDLER
+    ingress.metadata.namespace = 'default'
+    return ingress
+
+
+def test_remove_host_rule(unidler_ingress):
+    assert len(list(filter(
+        lambda rule: rule.host == HOSTNAME,
+        unidler_ingress.spec.rules))) == 1
+
+    unidler.remove_host_rule(HOSTNAME, unidler_ingress)
+
+    assert len(list(filter(
```

Comment:
Just curious: Do you need to convert to `list` in here?

Reply:
I was playing with the tests and I see that `filter()` actually returns a "filter object" therefore the need of the cast.

In theory we could get rid of the `len()` and write these assertions as:
```
assert list(filter(...))
[...]
assert not list(filter(...))
```

That's more compact but maybe terse :)

Reply:
👍